### PR TITLE
[MDS-5771] Adding display of prr or crr report type

### PR DESCRIPTION
**Pull request by asinn134**
## Objective 

[MDS-5771](https://bcmines.atlassian.net/browse/MDS-5771)

-Adding display of prr or crr report type in report page
-Currently should only have Code Required Report selected because the user selection in MDS-5781 hasn't been worked on yet.

Screenshot (263)

### CODE REVIEW

File: services/common/src/components/reports/ReportDetailsForm.tsx [hunk 1]
```diff
@@ -228,6 +229,19 @@ const ReportDetailsForm: FC<ReportDetailsFormProps> = ({
         initialValues={initialValues}
       >
         <Row gutter={[16, 8]}>
+          {system === SystemFlagEnum.core && (
+            <Col span={24}>
+              <Typography.Paragraph>What is the type of the report?</Typography.Paragraph>
```
Comment (taraepp) [inline services/common/src/components/reports/ReportDetailsForm.tsx]: Would recommend using the RenderRadio button instead of manually making it (and having the typography/required indicator not match...). Not keen on using 1/2 for values, hard to parse when you're debugging.

File: services/common/src/components/reports/ReportDetailsForm.tsx [hunk 2]
```diff
@@ -231,15 +242,22 @@ const ReportDetailsForm: FC<ReportDetailsFormProps> = ({
         <Row gutter={[16, 8]}>
           {system === SystemFlagEnum.core && (
             <Col span={24}>
-              <Typography.Paragraph>What is the type of the report?</Typography.Paragraph>
-              <Radio.Group
+              <Field
+                name="report_type"
+                id="report_type"
+                required
                 disabled={true}
```
Comment (asinn134) [inline services/common/src/components/reports/ReportDetailsForm.tsx:249]: I set disabled to true here because at the moment in Core we don't have the ability for the user to select a permit condition category code if they select PRR, my understanding is that will be worked on in MDS-5781. Should be able to adjust this to use `isEditMode` then

File: services/common/src/components/reports/ReportDetailsForm.tsx [hunk 3]
```diff
@@ -131,6 +137,11 @@ const ReportDetailsForm: FC<ReportDetailsFormProps> = ({
     }
   }, []);
 
+  useEffect(() => {
+    const reportType = initialValues?.permit_condition_category_code ? "PRR" : "CRR";
+    dispatch(change(FORM.VIEW_EDIT_REPORT, "report_type", reportType));
+  }, [!formValues?.report_type]);
```
Comment (asinn134) [inline services/common/src/components/reports/ReportDetailsForm.tsx:143]: I used `!formValues?.report_type` as the `useEffect` condition here because at the moment if a user discards the form changes and we are keeping `report_type` radio buttons disabled, the user cannot make a selection. This should be adjusted in the work done for MDS-5781

File: services/common/src/components/reports/ReportDetailsForm.tsx (hunk 1 above)
Comment (asinn134) [reply services/common/src/components/reports/ReportDetailsForm.tsx]: Thanks for the recommendation, this should be adjusted now.

**Review by henryoforeh-dev — APPROVED** [review]

**Review by taraepp — APPROVED** [review]

**Review by matbusby-fw — APPROVED** [review]
Nice work!
